docs/how-to-guides/data-basics/2-building-a-list-of-pages.md

```diff
@@ -0,0 +1,608 @@
+About. They could finish in a different order than they started. It is possible that _search=A_ would resolve after _
+search=About_ and we’d display the wrong data.
+
+Gutenberg data helps by handling the asynchronous part behind the scenes. `useSelect` remembers the most recent call and
+returns only the data we expect.
+
+Second, every keystroke would trigger an API request. If you typed About, deleted it, and retyped it, it would
+issue 10 requests in total even though we could reuse the data.
+
+Gutenberg data helps by caching the responses to API requests triggered by `getEntityRecords()`  and reuses them on
+subsequent calls. This is especially important when other components rely on the same entity records.
+
+All in all, the built-in utilities are designed to solve the typical problems so that you can focus on your application
+instead.
+
+## Step 5: Loading Indicator
+
+There is one problem with our search feature. We can’t be quite sure whether it’s still searching or showing no results:
+
+![](./media/list-of-pages/unclear-status.jpg)
+
+A few messages like  _Loading…_ or _No results_ would clear it up. Let’s implement them! First,  `PagesList` has to be
+aware of the current status:
+
+```js
+const { Spinner } = wp.components;
+function PagesList( { hasResolved, pages } ) {
+	if ( !hasResolved ) {
+		return <Spinner/>
+	}
+	if ( !pages?.length ) {
+		return <div>No results</div>
+	}
+	// ...
+}
+
+function MyFirstApp() {
+	// ...
+
+	return (
+		<div>
+			// ...
+			<PagesList hasResolved={ hasResolved } pages={ pages }/>
+		</div>
+	)
+}
+```
+
+<details><summary>Compiled snippet</summary>
+
+```js
+function PagesList( { hasResolved, pages } ) {
+	if ( !hasResolved ) {
+		return wp.element.createElement( wp.components.Spinner );
+	}
+	if ( !pages?.length ) {
+		return wp.element.createElement( 'div', {}, 'No results' );
+	}
+	// ...
+}
+
+function MyFirstApp() {
+	// ...
+
+	return (
+		wp.element.createElement( 'div', {},
+			// ...
+			wp.element.createElement( PagesList, { hasResolved, pages } ),
+		)
+	)
+}
+```
+
+</details>
+
+Note that instead of building a custom loading indicator, we took advantage of
+the [Spinner](https://developer.wordpress.org/block-editor/reference-guides/components/spinner/) component.
+
+We still need to know whether the pages selector `hasResolved` or not. We can find out using
+the  `hasFinishedResolution` selector:
+
+`wp.data.select('core').hasFinishedResolution( 'getEntityRecords', [ 'postType', 'page', { search: 'home' } ] )`
+
+It takes the name of the selector and the arguments and returns either `true` if the data was already loaded or `false`
+it we’re still waiting. Let’s add it to `wp.data.useSelect`:
+
+```js
+function MyFirstApp() {
+	// ...
+	const { pages, hasResolved } = wp.data.useSelect( select => {
+		// ...
+		return {
+			pages: {/* ... */ },
+			hasResolved:
+				select( wp.coreData.store ).hasFinishedResolution( 'getEntityRecords', ['postType', 'page', query] ),
```

Comment:
I'd love for to show the `getEntityRecords` selector here and extract the arguments into a variable in order that you can use it across both selector and resolver. In fact it would be even better to start out with the hard coded form and then show a refactor whereby you move to a variable for convenience. 

Why? 

1. Because it's super easy to have a resolver that doesn't "work" because you didm't pass the exact same arguments.
2. The pattern is well established in Gutenberg Core and thus pretty standard.

Reply:
Such a great observation @getdave, thank you! I just introduced that refactor – would you mind reviewing if that's what you meant?